src/user.cpp

```diff
@@ -245,6 +247,27 @@ OptionalMessage User::ExModeCommand(const Event& event){
 	utils::PrintStringVector(event.get_command_params());
 	return OptionalMessage::Empty();
 }
+
+OptionalMessage User::ExQuitCommand(const Event& event){
+	if (event.get_fd() == this->get_fd()) {
+		this->is_delete_ = true;
+	} else {
+		const User& executer = event.get_executer();
+		std::string prefix_message = executer.get_nick_name() + " QUIT : ";
+		for (std::vector<const Channel*>::iterator it =
+		this->joining_channels_.begin();
+		it != this->joining_channels_.end();
+		it++){
+			if((*it)->ContainsUser(executer)) {
+				std::string context_message = event.get_command_params().empty() ? "client quit" : event.get_command_params()[0];
+				if (event.get_event_type() == POLL_HUP)
+					context_message = "client dies and EOF occurs on socket";
+				return OptionalMessage::Create(this->fd_, prefix_message + context_message + "\r\n");
+			}
+		}
+	}
+	return OptionalMessage::Empty();
```

Comment:
{}でインデントを積みすぎると醜くなるので、避けるべきです。
ここではearly returnしても良いでしょう。
他にも気になる部分に修正を加えています。
```suggestion
	if (event.get_fd() == this->get_fd()) {
		this->is_delete_ = true;
		return OptionalMessage::Empty();
	}
	const User& executer = event.get_executer();
	std::string prefix_message = executer.get_nick_name() + " QUIT : ";
	for (std::vector<const Channel*>::iterator it =
	this->joining_channels_.begin();
	it != this->joining_channels_.end();
	++it) {
		if((*it)->ContainsUser(executer)) {
			std::string context_message;
			if (event.get_event_type() == POLL_HUP)
				context_message = "client dies and EOF occurs on socket";
			else if (event.get_command_params().empty())
				context_message = "client quit";
			else
				context_message = event.get_command_params()[0];
			return OptionalMessage::Create(this->fd_, prefix_message + context_message + "\r\n");
		}
	}
	return OptionalMessage::Empty();
```